### DISCUSSION TITLE
UMinho covers

### DISCUSSION THREAD

**Discussion by joaomlourenco** [joaomlourenco/novathesis#388] [Q&A] [answered]
Dear UMinho comunity,

I remade the covers for UMinho so that, in the final version, they are shown as required by UMinho services: one big A3-like page in landscape mode, with (from left to right) back-cover, book spine, front cover; and another big coloured page in grey or red(ish).

Can you please check if these covers are (probably) ok?  Thanks.
[capasMSc.pdf](https://github.com/joaomlourenco/novathesis/files/10098358/capasMSc.pdf)
[capasPhD.pdf](https://github.com/joaomlourenco/novathesis/files/10098359/capasPhD.pdf)

**Comment by BrunoSilvaUM** [comment]
Good afternoon,

Firstly, let me thank you for doing this project,. I am currently using it and has been helping me a lot on writing my thesis!

I believe that the cover must be as described in this pdf:

[CAPA-Dissertacao_MESTRADO.pdf](https://github.com/joaomlourenco/novathesis/files/10260380/CAPA-Dissertacao_MESTRADO.pdf)

The cover is probably ok when we set the config to the "final" version, the "working" version uses the old cover (at least for me).

Also, in the new cover the advisor and course are not specified. So I believe that the cover described on page 6 of that pdf must come after, but I'm not sure.

Another thing that happened to me was the problem described in https://github.com/joaomlourenco/novathesis/discussions/277.

Best regards!

**Reply by joaomlourenco** [reply]
@BrunoSilvaUM

First of all, let me tell you that @b-pereira had a big role in the support for UMinho. :)

I followed that document, but would like some feedback on the current  result.  I think you are right and the "second cover" should be there… if it is not it is my fault.  i'll fix it.

Th working version is exactly that, a working version, and uses conventional covers in A4 paper.  Once you change it to "provisional" or "final", the new A3 cover will be produced instead.

Concerning the problem described in #277, maybe it came back to life…. ;).  Which version are you using?

**Reply by BrunoSilvaUM** [reply]
Ok, thank you!
I believe that for the version 6.10.1, the problem #277 doesn't exist but it also doesn't have the A3 cover for the final version of the document.
The version that is available on the master branch right now has the A3 cover but I get the #277 problem.

Best regards!

**Reply by joaomlourenco** [reply]
There was a missing "%" after the "}" in line 1724 of file `novathesis.cls`.  Fixing it.

**Comment by joaomlourenco** [comment]
Fixes for both the missing "second cover" and the bug #227 were committed to the repository.

**Comment by Marcelo-Alves7** [comment]
Hello professor,

Thanks to you and @b-pereira for your effort in this work.

A while ago I had already raised this issue here in the discussion, and together with @b-pereira, we saw two approaches, a little more manual to create covers according to UMinho.
The covers you made for the final version are correct and in line with the services:
one big A3-like page in landscape mode, with (from left to right) back + spine + front together; and another big colored page in red(ish).

I'm using the latest version in Overleaf, [6.10.17], but the covers are still as they used to be, in A4.
The idea for the final version is to include the covers in a "manual" way, as mentioned here -> https://github.com/joaomlourenco/novathesis/discussions/361
or already possible to access automatically?

Thank you both.

**Reply by joaomlourenco** [reply, marked as answer]
@Marcelo-Alves7
The covers in A3 are available in branch 7.0.0, and will become available when that branch goes mainstream.
If you want to try it, checkout branch 7.0.0… but please note that there are some small incompatibilities with version(s) 6.x.y… but I'm sure you can handle it!  :)
If you try it, please give some feedback here! Thank you.

**Comment by Marcelo-Alves7** [comment]
Hello professor,
I haven't had a chance to check branch 7.0.0 Yet ... sorry I couldn't give feedback Yet.
I have been finishing the thesis revisions and this final phase is not easy, since there are still a few things !!
I would like to ask if this branch is expected to be available on the mainstream this month?
This is because I intend to deliver the thesis in April and would like to already count on the correct covers for UMinho template.
In any case, what is your advice to apply this branch and limit possible incompatibilities?
Thank You

**Reply by joaomlourenco** [reply]
Working on double checking the UMinho in v7.0.0.  More news later today.

**Reply by Marcelo-Alves7** [reply]
Hello professor,
I've been testing the 7.0.0 branch and I got some errors that I'm not able to fix.
I did the usual configuration for the Uminho template (https://github.com/joaomlourenco/novathesis/wiki/Uminho-Configuration), but I'm getting the following errors not expected:
![image](https://user-images.githubusercontent.com/98588105/225297087-32dad3ed-2183-4cb3-94cc-6236fa85ecce.png)

Can you help me identify the problems?

I will look forward to your related news regarding other incompatibilities and the issue of funding logos ([@#410](https://github.com/joaomlourenco/novathesis/discussions/410#discussioncomment-5312400))
Thank you very much

**Comment by joaomlourenco** [comment]
@Marcelo-Alves7 
Just pushed some corrections to UMINHO in branch 7.0.0.
Also added support for sponsor logos in the back cover.
Can you please try it again.

**Reply by Marcelo-Alves7** [reply]
Professor, I already had the opportunity to test.
Now the errors presented here (https://github.com/joaomlourenco/novathesis/discussions/388#discussioncomment-5320607) no longer appear to me, however, there seems to be a defect with the cover and back-cover, which in the previous version, despite of errors, appeared in the correct form.
The results are the following for docstatus=final (I also put it with the active sponsors)
![image](https://user-images.githubusercontent.com/98588105/225432255-aaf5a129-3ba9-4236-8714-54bdd24d4723.png)

**Reply by joaomlourenco** [reply]
Are you compiling with `make`?? If you are not, please do!
If you can't use `make`, please use `latexmk -shell-escape -synctex=1 -file-line-error  -lua template`.
Please let me know if it worked for you.

**Reply by Marcelo-Alves7** [reply]
Professor, I'm compiling with Overleaf.
I'm not familiar with `make` or `latexmk`.
Is it some configuration file to control compilation?
I did a little research using this initialization with Overleaf, but I couldn't figure out how to apply it.
I created a `latexmkrc.tex` file and pasted the command line, but I don't think that's it...
![image](https://user-images.githubusercontent.com/98588105/225466542-74fb808b-269a-4836-b68d-67223d2dba32.png)

Any suggestion?

**Comment by joaomlourenco** [comment]
Did you change the compiler in overleaf to xelatex or lualatex?
Did you upload the font files to the proper location?

**Reply by Marcelo-Alves7** [reply]
yes, professor.  I did the usual configuration for the Uminho template, namely the compiler and I include the font files properly.
It doesn't return any errors, just warnings.
But I noticed one detail:
If I use the xelatex compiler, the cover and back cover format appears like this -https://github.com/joaomlourenco/novathesis/discussions/388#discussioncomment-5327592
However, if I change the compiler to lualatex, the result is different and seems correct to me:
![image](https://user-images.githubusercontent.com/98588105/225582688-17f2312d-9868-4aa7-aef5-917719446584.png)
Maybe it's better to change the sponsor logos to white since we have a gray background on the back cover. Does the template allow this change?
I'll try to do some more tests.
Thanks

**Reply by joaomlourenco** [reply]
I confirm that the the covers for “final” mode are failing with XeTeX.  Will investigate.

**Reply by Marcelo-Alves7** [reply]
Thanks professor.
If you discover any more conflicts, let me know, and I'll try to test them too.

A brief question, to leave the template ready to be able to migrate:
- If I want to add the sponsor logos in white since we have a gray background on the back cover (UMinho asks this way), do I only need to put the new images in the `NOVAthesisFiles\Images` folder and then update the customization sponsors in `3_Config`?

**Reply by joaomlourenco** [reply]
You can put the logos in `Chapters/Figures` or in `NOVAthesisFiles\Images`, as you prefer.  But I suggest you add them to `Chapters/Figures`. Then update the names in `Config/3_cover.tex`.

**Reply by Marcelo-Alves7** [reply]
Thank you professor.
I'm going to prepare to migrate the template version.
If you find any more news regarding the build, please let me know.

**Reply by joaomlourenco** [reply]
The covers for uminho now work with both XeLaTeX and LuaLaTeX.
